internal/testrunner/runners/system/runner.go

```diff
@@ -519,14 +529,50 @@ func checkEnrolledAgents(client *kibana.Client, ctxt servicedeployer.ServiceCont
 		return true, nil
 	}, 5*time.Minute)
 	if err != nil {
-		return nil, errors.Wrap(err, "agent enrollment failed")
+		return nil, fmt.Errorf("agent enrollment failed: %w", err)
 	}
 	if !enrolled {
 		return nil, errors.New("no agent enrolled in time")
 	}
 	return agents, nil
 }
 
+func installPackage(client *kibana.Client, manifest *packages.PackageManifest, rootPath string) error {
```

Comment:
This looks similar to the factory created in `cmd/install.go` file. Do you think it would be possible to reuse some of the code? Maybe the installer code: 
- for local packages: internal/packages/installer/installer.go
- for zip packages: internal/packages/installer/zip_installer.go

https://github.com/elastic/elastic-package/tree/main/internal/packages/installer

Reply:
I have merged both codes in a single factory in the `installer` package, let me know what you think.

I am also passing the kibana client from the callers, I would like to have more explicit initializations of the clients we use. This is something I have WIP to be able to get the settings from the profile in this branch https://github.com/elastic/elastic-package/compare/main...jsoriano:elastic-package:clients-from-profile?expand=1.

Reply:
> I have merged both codes in a single factory in the installer package, let me know what you think.

It looks perfect! :muscle: thanks for doing this change!


> I am also passing the kibana client from the callers, I would like to have more explicit initializations of the clients we use. This is something I have WIP to be able to get the settings from the profile in this branch https://github.com/elastic/elastic-package/compare/main...jsoriano:elastic-package:clients-from-profile?expand=1.

Ok! that would also help us to add more tests if the kibana client is set as a parameter :+1: